### DISCUSSION TITLE
How to set up borg to connect to davmail and exchange?

### DISCUSSION THREAD

**Discussion by tps800** [mikeberger/borg_calendar#187] [General]
Since I could not find any hint: how do I have to set up Borg to connect to Davmail and exchange?

**Comment by mikeberger** [comment]
Caldav setup is literally trial and error and doesn't work with some servers due to limitations of the third-party caldav library that I use.

I have no way to test davmail myself. If I had to guess based on the information in the davmail FAQ, I'd try these settings under the Borg Ical Options for CALDAV:

CALDAV Server: the name of your server, like xxx.com or localhost:8080
Server Base Path: /
CALDAV User: your email (I think)
CALDAV Password: your password
CALDAV Calendar: the name of your calendar
Principal Path: principals/users
User Path: users

There is about a 10% chance of this working

**Comment by tps800** [comment]
Seems to not work this way.

**Thunderbird:**
User: username@domain
Password: password
Address: http://localhost:1080/users/<mailaddress>/calendar

**borg:**
Server: localhost:1080
Server Base Path: /
User: user@domain
Password: password
Calendar: calendar
Addressbook: default
Principal Path:
CALDAV User Path: users

leads to:
```
Syncing....Please wait
SYNC: connect to http://localhost:1080/
net.fortuna.ical4j.connector.ObjectStoreException: net.fortuna.ical4j.connector.FailedOperationException: Principals not found at [/users/user@domain]
```

If I define:
Server: localhost:1080
Server Base Path: /users
User: user@domain
Password: password
Calendar: calendar
Addressbook: default
Principal Path: user@maildomain
CALDAV User Path: calendar

Borg tries to find calendars at:
`[/users/user@maildomain/user@domain]`

what is wrong too.
I could not find any working combination.

Within:
`SYNC: connect to http://localhost:1080` -> (no slash at the end)

and

`[/users/<mailaddress>/calendar]`

Would do what is expected.
So what is what here? And where do I put the mailaddress if Username is not equal to mailaddress?

**Comment by mikeberger** [comment]
Did you try it this way?

Server Base Path: /
CALDAV User: your email (I think)
CALDAV Password: your password
CALDAV Calendar: the name of your calendar
Principal Path: principals/users
User Path: users

Unfortunately, the library I use to connect to caldav is a bit buggy. Even the error message "Principals not found at [/users/user@domain]" that comes out is wrong. The code mistakenly prints the "user" path in the brackets instead of the principal path, so you can't really tell what path the code is trying to use to connect.

FYI - I use this library to connect to caldav: https://github.com/ical4j/ical4j-connector

The version that I use is buggy. The developer has been working on a new version for some time, but it is not available yet.

I don't have an alternative to the ical4j-connector, so I am just waiting for a newer version from the developer to see if it fixes the problems.
